ms2deepscore/MS2DeepScoreData.py

```diff
@@ -0,0 +1,166 @@
+        self.d_bins = set_d_bins_fixed(number_of_bins, mz_min=mz_min, mz_max=mz_max)
+        self.peak_scaling = peak_scaling
+        self.allowed_missing_percentage = allowed_missing_percentage
+        self.peak_to_position = None
+        self.known_bins = None
+        self.generator_args = {}
+        self.spectrums_binned = None
+        self.inchikeys_all = None
+
+    def collect_binned_spectrums(self, spectrums: list, progress_bar=True):
+        """Create 'vocabulary' of bins that have peaks in spectrums.
+        Derive binned spectrums from spectrums and store them.
+
+        Parameters
+        ----------
+        spectrums
+            List of spectrums.
+        progress_bar
+            Show progress bar if set to True. Default is True.
+        """
+        print("Collect spectrum peaks...")
+        peak_to_position, known_bins = unique_peaks_fixed(spectrums, self.d_bins, self.mz_min)
+        print(f"Calculated embedding dimension of {len(known_bins)}.")
+        self.peak_to_position = peak_to_position
+        self.known_bins = known_bins
+
+        print("Convert spectrums to binned spectrums...")
+        self.spectrums_binned = self.create_binned_spectrums(spectrums, progress_bar)
+
+        # Collect inchikeys
+        self._collect_inchikeys(spectrums)
+
+    def create_binned_spectrums(self, input_spectrums: list, progress_bar=True):
+        """Create binned spectrums from input spectrums.
+
+        Parameters
+        ----------
+        input_spectrums
+            List of spectrums.
+        progress_bar
+            Show progress bar if set to True. Default is True.
+
+        Returns: List[dict]
+            List of binned spectrums created from input_spectrums.
+        """
+        spectrums_binned, missing_fractions = create_peak_list_fixed(input_spectrums,
+                                                                     self.peak_to_position,
+                                                                     self.d_bins, mz_min=self.mz_min,
+                                                                     peak_scaling=self.peak_scaling)
+        spectrums_binned_dicts = []
+        for i, spec in enumerate(tqdm(spectrums_binned, disable=(not progress_bar))):
+            assert 100*missing_fractions[i] <= self.allowed_missing_percentage, \
+                f"{100*missing_fractions[i]:.2f} of weighted spectrum is unknown to the model."
+            spectrums_binned_dicts.append(create_peak_dict(spec))
+        return spectrums_binned_dicts
+
+    def _collect_inchikeys(self, spectrums):
+        """Read inchikeys from spectrums and create inchkeys array.
+
+        Parameters
+        ----------
+        spectrums
+            List of spectrums.
+        """
+        inchikeys_list = []
+        for s in spectrums:
+            inchikeys_list.append(s.get("inchikey"))
+
+        self.inchikeys_all = np.array(inchikeys_list)
+
+    def set_generator_parameters(self, **settings):
```

Comment:
I am not sure what you plan to do with this exactly, but this doesn't feel very pythonic. These are parameters for the data generator, so I think they should be set there. 2 suggestions for how to do this in a different way:

## 1
* Pass spectrums in a `spectrums: List[Spectrum]` argument to a DataGenerator class
* Pass `spectrum_binner: SpectrumBinner` as argument to DataGenerator class
* Call `self.spectrum_binner.fit` method in on of the DataGenerator methods

# 2
* Bin the data using `SpectrumBinner` in a step before DataGenerator.
* Pass already-binned data to argument `binned_spectrums` for DataGenerator class (and possibly others like `known_bins`)

Reply:
True. I had a bit too much an everything-about-this-data object in mind originally, but if it becomes `SpectrumBinner` then this argument settings won't belong here.

Reply:
Cool!